SocketRocket/SRWebSocket.m

```diff
@@ -145,6 +145,122 @@ @implementation SRWebSocket {
     
     NSArray<NSString *> *_requestedProtocols;
     SRIOConsumerPool *_consumerPool;
+
+    //Proxy support
+    NSString *_httpProxyHost;
+    uint32_t _httpProxyPort;
+    BOOL _connectingToProxy;
+
+}
+
+// get proxy setting from device setting
+-(void) configureProxy
+{
+    SRFastLog(@"configureProxy");
+    NSDictionary *proxySettings = CFBridgingRelease(CFNetworkCopySystemProxySettings());
+    NSArray *proxies = CFBridgingRelease(CFNetworkCopyProxiesForURL((__bridge CFURLRef)_url, (__bridge CFDictionaryRef)proxySettings));
+    if (proxies.count == 0) {
+        SRFastLog(@"configureProxy no proxies");
+        [self openConnection];
+        return;                 // no proxy
+    }
+    NSDictionary *settings = [proxies objectAtIndex:0];
+    NSString *proxyType = settings[(NSString *)kCFProxyTypeKey];
+    if ([proxyType isEqualToString:(NSString *)kCFProxyTypeAutoConfigurationURL]) {
+        NSURL *pacURL = settings[(NSString *)kCFProxyAutoConfigurationURLKey];
+        if (pacURL) {
+            [self fetchPAC:pacURL];
+            return;
+        }
+    }
+    if ([proxyType isEqualToString:(NSString *)kCFProxyTypeHTTP] || [proxyType isEqualToString:(NSString *)kCFProxyTypeHTTPS]) {
+        _httpProxyHost = settings[(NSString *)kCFProxyHostNameKey];
+        _httpProxyPort = [settings[(NSString *)kCFProxyPortNumberKey] intValue];
+    }
+    if (_httpProxyHost) {
+        SRFastLog(@"SRWebSocket using proxy %@:%u", _httpProxyHost, _httpProxyPort);
+    }
+    [self openConnection];
+}
+
+- (void)fetchPAC:(NSURL *)PACurl
+{
+    SRFastLog(@"SRWebSocket fetchPAC:%@", PACurl);
+
+	if ([PACurl isFileURL]) {
+        NSError *nsError = nil;
+        NSString *script = [NSString stringWithContentsOfURL:PACurl
+                                                usedEncoding:NULL
+                                                       error:&nsError];
+
+        if (nsError) {
+            [self openConnection];
+            return;
+        }
+        [self runPACScript:script];
+		return;
```

Comment:
Nit: Looks like this is missing indentation - if you could run `auto-indent` in Xcode on all of the new code - that would help tremendously.